### ISSUE TITLE
10-bit color appears to be working, but the actual video signal seems to be 8-bit.

### ISSUE THREAD

**Issue by joker1007** [hyprwm/Hyprland#8706]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland 0.45.2 built from branch  at commit 12f9a0d0b93f691d4d9923716557154d74777b0a  ([gha] Nix: update inputs).
Date: Tue Nov 19 21:47:18 2024
Tag: v0.45.2, commits: 5451
built against aquamarine 0.5.0


flags set:
debug


System Information:
System name: Linux
Node name: durandal
Release: 6.12.3-gentoo
Version: #1 SMP PREEMPT_DYNAMIC Mon Dec  9 17:04:18 JST 2024


GPU information: 
0b:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] [1002:73bf] (rev c0) (prog-if 00 [VGA controller])


os-release: NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
VERSION_ID="2.17"


plugins:
  hyprexpo by Vaxry ver 1.0
```

</details>

### Description

I set the bitdepth to 10 and started Hyprland.
There were no problems with startup and the image appears on the display.
The format was also recognized as 10bit in `hyprctl monitors`.

However, the indicator on the display showed the color depth as 8 bps. (The display I am using is an ASUS XG32UQ, which has an indicator showing the current color depth.)

Other devices that receive HDMI signals also confirmed that the actual signal being received was 8 bits.

I set `render_bit_depth` to 10 in sway-1.10 and started it up, and sway is correctly sending a 10-bit signal as far as I can see in the indicator.

In other words, I think that Hyprland is not correctly transmitting 10-bit video signals.

I visually compared the images generated by https://github.com/jursonovicst/gradient.
The indicator on the display showed that the image was recognized as 8-bit, but to my eyes it looked like it was actually a 10-bit output.

In reality, this may not be a problem, but the behavior seems strange.

### How to reproduce

set bitdepth, 10 like below.

```
monitor=DP-1, 3840x2160@120, 0x0, 1, bitdepth, 10
```

### Crash reports, logs, images, videos

[hyprland.log with HYPRLAND_TRACE=1 AQ_TRACE=1](https://github.com/user-attachments/files/18128567/hyprland.log)

**Comment by vaxerski** [comment]
```
[TRACE] [AQ] drm: Using drmModeAddFB2WithModifiers to import buffer into KMS: Size [Vector2D: x: 3840, y: 2160] with format XR30 and mod 144115188622392067
```

All seems in ordnung from the logs. KMS gets an XR30 (XRGB2101010) buffer and accepts it.

I can't think of why your display would do that, maybe it somehow samples the cursor plane? (which would be 8 bit yes) - though those shouldnt be visible to the display. IDK, maybe the driver is drunk.

Anyways, if to your eyes it's 10b, I think it might just be a bug in reporting the bitness.

**Comment by joker1007** [comment]
@vaxerski 
Thank you very much for your response!

I am not familiar with the mechanics of display rendering, but I wonder if some bug could bury the difference between the actual signal and the reported color depth bit.

Anyway, The logs indicate that the rendering itself is likely to be done properly, so I trust my eyes.

Thank you again.
If there is nothing of concern from the maintainer's point of view, the issue can be closed.

**Comment by vaxerski** [comment]
closing then
